src/main/java/io/openliberty/tools/common/plugins/util/PrepareFeatureUtil.java

```diff
@@ -104,57 +110,68 @@ private Map<File, String> populateESAMap(File additionalBOM) {
+			version);
+		File generatedJson = generateJson(targetJsonFile, esaMap);
+		if (generatedJson.exists()) {
+		    jsonFile = generatedJson;
+		    provideJsonFileDependency(generatedJson, groupId, version);
+		    info("The features.json has been generated at the following location: " + generatedJson);
 		}
-
+	    } catch (PluginExecutionException e) {
+		warn(e.getMessage());
+	    }
+	}
 
 	/**
-	 * Download the Artifacts mentioned within the additionalBOM pom file
+	 * Download the Artifacts mentioned within the additionalBOM pom file.
+	 * Required artifact properties are "groupId, artifactId, version and type".
+	 * If any of the properties are missing, then it will throw NullPointerException. 
 	 * 
 	 * @param additionalBOM The BOM file
 	 * @return A map of Files to groupIds
 	 * @throws PluginExecutionException throws error if unable to download the
 	 *                                  artifacts
 	 */
 	private Map<File, String> downloadArtifactsFromBOM(File additionalBOM) throws PluginExecutionException {
-		Map<File, String> result = new HashMap<File, String>();
-		try {
-			DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
-			DocumentBuilder db = dbf.newDocumentBuilder();
-			Document doc = db.parse(additionalBOM);
-			doc.getDocumentElement().normalize();
-			NodeList dependencyList = doc.getElementsByTagName("dependency");
-			for (int itr = 0; itr < dependencyList.getLength(); itr++) {
-				Node node = dependencyList.item(itr);
-				if (node.getNodeType() == Node.ELEMENT_NODE) {
-					Element eElement = (Element) node;
-					String groupId = eElement.getElementsByTagName("groupId").item(0).getTextContent();
-					String artifactId = eElement.getElementsByTagName("artifactId").item(0).getTextContent();
-					String version = eElement.getElementsByTagName("version").item(0).getTextContent();
-					String type = eElement.getElementsByTagName("type").item(0).getTextContent();
-
-					File artifactFile = downloadArtifact(groupId, artifactId, type, version);
-					result.put(artifactFile, groupId);
-				}
+	    Map<File, String> result = new HashMap<File, String>();
+	    String[] MavenCoord =  {"groupId", "artifactId", "type", "version"};
+	   
+	    try {
+		DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+		DocumentBuilder db = dbf.newDocumentBuilder();
+		Document doc = db.parse(additionalBOM);
+		doc.getDocumentElement().normalize();
+		NodeList dependencyList = doc.getElementsByTagName("dependency");
+		for (int itr = 0; itr < dependencyList.getLength(); itr++) {
+		    Node node = dependencyList.item(itr);
+		    if (node.getNodeType() == Node.ELEMENT_NODE) {
+			Element eElement = (Element) node;
+			// error checking
+			for (String tag : MavenCoord) {
+			    Node tmp_node = eElement.getElementsByTagName(tag).item(0);
+			    if (tmp_node == null) {
+				throw new PluginExecutionException(
+					"Error: <" + tag + "> tag nof found in BOM file.");
```

Comment:
There is a typo and I think this could use a better message. For example, what BOM file was it trying to process? But instead of having this extra `for` loop checking for null, you could just check that the result of `item(0)` was not null before calling `getTextContent()` at lines 161-165.